Pizza Project, Pair Programming, Jihyun An and Lisa Sturegard

LisaStur opened the pull request:
https://musing-swanson-3769c1.netlify.com/

We struggled to start with, but with the Monday tutorial with Van Taylor it all became a lot clearer.

Things we would like to have clarified or explained in more detail:
Would be nice to work more with loop.
And conditional statement/If-statement to check the data type.

LisaStur committed:
Iteration One Pizza Project

LisaStur committed:
Iteration Two Pizza Project

LisaStur committed:
Third interation Pizza Project

LisaStur committed:
Fourth iteration Pizza Project

LisaStur committed:
Iteration 4 Project Pizza

LisaStur committed:
Iteration 5 Pizza Project

LisaStur committed:
Iteration 6 Pizza Project

LisaStur committed:
Iteration 7 Pizza Project

LisaStur committed:
Update README.md

LisaStur committed:
Updated with more versions for prompt entry and a cheat version on going back to the prompt

ericaglimsholt commented on code/pizzabot.js:
```diff
@@ -4,4 +4,88 @@ const pepperoni = "Pepperoni Pizza"
 const pizzaPrice = 80
 
-//Put your Javscript code here:
+var x = vegetarian
```
Try to use `const` or `let` before using `var` when declaring a variable. In this case `const` would suit perfect! Another thing is the naming could be better. A name should be a description of what it is. Like in this example `x` doesn't say that it is a vegetarian pizza. 
```suggestion
const vegetarian = "vegetarian pizza"
```

ericaglimsholt commented on code/pizzabot.js:
```diff
@@ -4,4 +4,88 @@ const pepperoni = "Pepperoni Pizza"
+var y = hawaiian
+var z = pepperoni
+
+alert("Hey! Happy to serve you pizza! On our menu we have " + x + ", " + y + " and " + z + ".");
```
You can make this line more readable with using `${}`!
```suggestion
alert("Hey! Happy to serve you pizza! On our menu we have ${x}, ${y} and ${z}.");
```

ericaglimsholt commented on code/pizzabot.js:
```diff
@@ -4,4 +4,88 @@ const pepperoni = "Pepperoni Pizza"
+if (orderName.toLowerCase() == "vegetarian") {
+  modifiedName = x;
+} else if (orderName.toLowerCase() == "vegetarian pizza") {
+  modifiedName = x;
```
When checking multiple possibilities you can use "or" with `||`. Example:
```
if(order == "vegetarian" || order == "vegetarian pizza") {
  //do something
}
```

ericaglimsholt commented on code/pizzabot.js:
```diff
@@ -4,4 +4,88 @@ const pepperoni = "Pepperoni Pizza"
+} else {
+  alert('Select a pizza from the menu');
+  let orderName = prompt("Enter the name of the pizza you want to order today.");
+  if (orderName.toLowerCase() == "vegetarian") {
```
Instead of doing this again you could create a function that you run again. Example:
```
const matchedOrder = false

const checkOrder = () => {
  // check order
  // return true if it matches
  matchedOrder = true
}

// if the matched order is false
if(!matchedOrder) {
  alert('Select a pizza from the menu')
  // run function for selecting pizza again
}
```

ericaglimsholt commented on code/pizzabot.js:
```diff
@@ -4,4 +4,88 @@ const pepperoni = "Pepperoni Pizza"
+const totalCost = orderQuantity * 80;
+
+if (orderQuantity < 3) {
+  cookingTime = "10";
```
When writing numbers you don't need `"` around. With `"` around it means that it's a string (text)
```suggestion
  cookingTime = 10;
```

ericaglimsholt commented on code/pizzabot.js:
```diff
@@ -4,4 +4,88 @@ const pepperoni = "Pepperoni Pizza"
+console.log(orderQuantity);
+console.log(`The customer has ordered ${orderQuantity} of ${orderName}.`);
+console.log('It will cost ' + orderQuantity * 80 + 'kr.');
+console.log('It will take ' + cookingTime + ' to prepare.')
```
Using `console.log()` is meant to be for developing purpose (so you can check so everything works) but you can delete it when you are making a pull request!

ericaglimsholt reviewed (approved):
Hi Lisa and Jihyun! I hope that you find the pair programming to be helpful when learning something new. I've added some comments for make your javascript better, but you managed to get everything that was needed for this assignment :)

LisaStur commented:
> Hi Lisa and Jihyun! I hope that you find the pair programming to be helpful when learning something new. I've added some comments for make your javascript better, but you managed to get everything that was needed for this assignment :)

Big thanks for your feedback Erica! We learnt a lot from the project, and even more with your comments! (We have updated the project to get some practice). We are still struggeling with the "getting-back-to the-oriinal-prompt", and have booked some 1-on-1 time for that this week, to help us get our heads around it).

LisaStur committed:
Realised I never pushed the latest version, with updated return-to-start if you don't get the Pizza-name right

LisaStur committed:
Update README.md

Updated link to site.
